### PR TITLE
35 support shortcut schema syntax

### PR DESCRIPTION
**Pull request by SleepWalker**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: index.js [hunk 1]
```diff
@@ -27,17 +27,24 @@ function validateResponse (fastify, opts, next) {
   function buildHook (schema) {
     const statusCodes = {}
     for (const statusCode in schema) {
-      statusCodes[statusCode] = ajv.compile(schema[statusCode])
+      const responseSchema = schema[statusCode]
+      statusCodes[statusCode] = ajv.compile(
+        // replicate getSchemaAnyway from fastify
+        // https://github.com/fastify/fastify/blob/23371945d01c270af24f4a5b7e2e31c4e806e6b3/lib/schemas.js#L113
+        !responseSchema.type && !responseSchema.properties
```
Comment (climba03003) [inline index.js:34]: Can you copy the whole function? `oneOf`, `anyOf`, `allOf` ... are supported in the original function.

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (SleepWalker) [reply index.js:34]: Do you mean `ajv.compile` function?

```js
statusCodes[statusCode] = !responseSchema.type && !responseSchema.properties ? ajv.compile(...) : ajv.compile(...);
```

Or do you mean entire buildHook? I haven't also understood about `oneOf` and other.

Comment (climba03003) [reply index.js:34]: Yes. the `ajv.compile` function.
You are using a condition statement for the `schema` which do not support `oneOf`, `anyOf`, etc.

Comment (SleepWalker) [reply index.js:34]: Ah, I see. I've skipped this part of logic:

image

File: index.js [hunk 2]
```diff
@@ -51,10 +58,10 @@ function validateResponse (fastify, opts, next) {
 }
 
 function schemaErrorsText (errors) {
-  var text = ''
-  var separator = ', '
-  for (var i = 0; i < errors.length; i++) {
-    var e = errors[i]
+  let text = ''
+  const separator = ', '
+  for (let i = 0; i < errors.length; i++) {
```
Comment (climba03003) [inline index.js:63]: Keep the `var` here for performance.
```suggestion
  for (var i = 0; i < errors.length; i++) {
```

Comment (SleepWalker) [reply index.js:63]: interesting :thinking:

File: index.js (hunk 1 above)
Comment (SleepWalker) [reply index.js]: I've copied an entire function from `fastify`

**Review by climba03003 — APPROVED** [review]
LGTM.

**Review by Eomm — COMMENTED** [review]
I forgot to submit the review 😂 

Adding a possible improvement 👍🏽

File: test.js [hunk 3]
```diff
@@ -39,6 +39,38 @@ test('Should return a validation error', async t => {
   })
 })
 
+test('Should support shortcut schema syntax', async t => {
+  const fastify = Fastify()
```
[the remark this answers is not in the document]
Comment (SleepWalker) [reply test.js:43]: Oh, I've seen that flag but I've thought that in order to support it, it should be added as a separate option for `fastify-response-validation` and that looked to me as unnecessary complication.

If we can inherit root fastify config it looks like a good addition. Should I submit a separate PR for that?

[the remark this answers is not in the document]
Comment (Eomm) [reply test.js:43]: If you would like, it would be really appreciated!

**Pull request merged**